File: swift/problem686.swift
import Foundation

class Solution {
    func repeatedStringMatch(_ A: String, _ B: String) -> Int {
        if !Set(B).isSubset(of: Set(A)) {
            return -1
        }
        
        var s = String(repeating: A, count: B.count/A.count)
        repeat {
            if s.contains(B) {
                return s.count/A.count
            } else {
                s += A
            }
        } while (s.count < B.count + 2*A.count)
        return -1
    }
}

File: python/problem686.py
class Solution:
    def repeatedStringMatch(self, A: str, B: str) -> int:
        s = A
        k = 1
        if len(s) >= len(B):
            while k<4:   
                if B in s:
                    return k
                s += A
                k += 1
            return -1
                
                
        while len(s) < len(B):
            s += A
            k += 1
        m = k
        while m < 2*k:
            if B in s:
                return m
            s += A
            m += 1
            
        return -1